src/pages/ja/editor-setup.md

```diff
@@ -0,0 +1,41 @@
+---
+layout: ~/layouts/MainLayout.astro
+setup: |
+  import Badge from '~/components/Badge.astro';
+title: エディタのセットアップ
+description: Astroで開発するため、エディタの設定をしましょう。
+i18nReady: true
+---
+
+エディタをカスタマイズし、開発者体験を向上させ新機能をアンロックします。
+
+## VS Code
+
+[VS Code](https://code.visualstudio.com/) はMicrosoft社が開発した、web開発者に人気のあるコードエディタです。 VS Codeのエンジンは [GitHub Codespaces](https://github.com/features/codespaces) や [Gitpod](https://gitpod.io/)といった人気のあるブラウザ内コードエディタもサポートしています。
+
+Astro はどのようなコードエディタでも動作しますが、VS CodeはAstroで開発する際におすすめのエディタです。 私たちはいくつかの重要な機能のアンロックと、開発者体験を向上させる公式[VS
+ Code 拡張機能](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode)をメンテナンスしています。
+
+- `.astro`ファイルのシンタックスハイライト
+- `.astro`ファイルのTypeScript型情報
+- [VS Code Intellisense](https://code.visualstudio.com/docs/editor/intellisense)によるコード補完、ヒントなど
+
+早速、[Astro VS Code Extension](https://marketplace.visualstudio.com/items?itemName=astro-build.astro-vscode) をインストールしてみましょう。
+
+
+>⚙️ Astroプロジェクトでどのように[TypeScriptをセットアップ](/en/guides/typescript/)するのか見る。
```

Comment:
```suggestion
>⚙️ Astroプロジェクトでどのように[TypeScriptをセットアップ](/ja/guides/typescript/)するのか見る。
```
Just a link to correct, because even when the page doesn't yet exist, we still use the language in the URL to keep the reader in the translated language version of the site.  :)